Dev Container outdated golang version

<!--
Template relevant to bug reports only!

Keep issue title verbose enough and add prefix telling
about what components it touches e.g "query:" or ".*:"
-->

<!--
In case of issues related to exact bucket implementation, please ping corresponded maintainer from list here: https://github.com/thanos-io/thanos/blob/main/docs/storage.md
-->

**Thanos, Prometheus and Golang version used**:

<!--
Output of "thanos --version" or docker image:tag used.
(Double-check if all deployed components/services have expected versions)

If you are using custom build from main branch, have you checked out the tip of the main?
-->

**Object Storage Provider**: N/A

**What happened**: When develop using `Open in Github Codespaces`, and then redirected to VSCode, the codespace fails to build because mismatched go version specified in `.devcontainer` and in `go.mod`

**What you expected to happen**: There should be no error setting up the project in codespaces

**How to reproduce it (as minimally and precisely as possible)**:  Click on [![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://github.com/codespaces/new?hide_repo_select=true&ref=main&repo=109162639), and then create a codespace.

**Full logs to relevant components**:

```
=================================================================================
2024-12-31 17:53:33.573Z: Running blocking commands...
2024-12-31 17:53:33.598Z: $ devcontainer up --id-label Type=codespaces --workspace-folder /var/lib/docker/codespacemount/workspace/thanos --mount type=bind,source=/.codespaces/agent/mount/cache,target=/vscode --user-data-folder /var/lib/docker/codespacemount/.persistedshare --container-data-folder .vscode-remote/data/Machine --container-system-data-folder /var/vscode-remote --log-level trace --log-format json --update-remote-user-uid-default never --mount-workspace-git-root false --omit-config-remote-env-from-metadata --skip-non-blocking-commands --expect-existing-container --config "/var/lib/docker/codespacemount/workspace/thanos/.devcontainer/devcontainer.json" --override-config /root/.codespaces/shared/merged_devcontainer.json --default-user-env-probe loginInteractiveShell --container-session-data-folder /workspaces/.codespaces/.persistedshare/devcontainers-cli/cache --secrets-file /root/.codespaces/shared/user-secrets-envs.json
2024-12-31 17:53:33.785Z: @devcontainers/cli 0.68.0. Node.js v18.20.5. linux 6.5.0-1025-azure x64.
2024-12-31 17:53:34.063Z: Running the onCreateCommand from devcontainer.json...

2024-12-31 17:53:34.064Z: make build
2024-12-31 17:53:34.513Z: go: go.mod requires go >= 1.23.0 (running go 1.22.10; GOTOOLCHAIN=local)
2024-12-31 17:53:34.513Z: make: *** [Makefile:171: deps] Error 1
2024-12-31 17:53:34.519Z: onCreateCommand failed with exit code 2. Skipping any further user-provided commands.

2024-12-31 17:53:34.519Z: Error: Command failed: /bin/sh -c make build
2024-12-31 17:53:34.520Z:     at b7 (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:235:130)
2024-12-31 17:53:34.520Z:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-12-31 17:53:34.521Z: {"outcome":"error","message":"Command failed: /bin/sh -c make build","description":"The onCreateCommand in the devcontainer.json failed.","containerId":"4ec9fc45ea0113e53b0f44d2c8995e6014a6ad8e33f7a9b00d6dcfce7e011ed6"}
2024-12-31 17:53:34.521Z:     at async Am (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:227:4483)
2024-12-31 17:53:34.521Z:     at async Xw (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:227:3828)
2024-12-31 17:53:34.521Z:     at async $w (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:227:2835)
2024-12-31 17:53:34.522Z:     at async fa (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:227:2438)
2024-12-31 17:53:34.522Z:     at async mtA (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:465:1534)
2024-12-31 17:53:34.523Z:     at async NH (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:465:964)
2024-12-31 17:53:34.523Z:     at async YtA (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:482:3771)
2024-12-31 17:53:34.524Z:     at async eB (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:482:4886)
2024-12-31 17:53:34.532Z: devcontainer process exited with exit code 1

====================================== ERROR ====================================
2024-12-31 17:53:34.536Z: Failed to create container.
=================================================================================
2024-12-31 17:53:34.536Z: Error: Command failed: /bin/sh -c make build
2024-12-31 17:53:34.540Z: Error code: 1302 (UnifiedContainersErrorFatalCreatingContainer)

====================================== ERROR ====================================
2024-12-31 17:53:34.554Z: Container creation failed.

```


<!--
Uncomment if you would like to post collapsible logs:

<details>Logs
<p>

```
```

</p>
</details>
-->

**Anything else we need to know**:

<!--
Uncomment and fill if you use not casual environment or if it might be relevant.

**Environment**:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Others:

-->

--- a/.devcontainer/Dockerfile
+++ b/.devcontainer/Dockerfile
@@ -1,5 +1,5 @@
 # For details, see https://github.com/devcontainers/images/tree/main/src/go
-FROM mcr.microsoft.com/devcontainers/go:1.22
+FROM mcr.microsoft.com/devcontainers/go:1.23
 
 RUN echo "Downloading prometheus..." \
     && curl -sSL -H "Accept: application/vnd.github.v3+json" "https://api.github.com/repos/prometheus/prometheus/releases" -o /tmp/releases.json \

